rest/api_test.go

```diff
@@ -9456,6 +9456,181 @@ func TestAttachmentDeleteOnExpiry(t *testing.T) {
 
 }
 
+// CBG-2143: Make sure the REST API is returning forbidden errors if unsupported config option is set
+func TestForceAPIForbiddenErrors(t *testing.T) {
+	const expectedErrorReason = "forbidden"
+	rt := NewRestTester(t, &RestTesterConfig{
+		SyncFn: `
+		function(doc, oldDoc) {
+			if (!doc.doNotSync) {
+				access("NoPerms", "chan2");
+				access("Perms", "chan2");
+				requireAccess("chan");
+				channel(doc.channels);
+			}
+		}`,
+		DatabaseConfig: &DatabaseConfig{DbConfig: DbConfig{
+			Unsupported: &db.UnsupportedOptions{
+				ForceAPIForbiddenErrors: true,
```

Comment:
Do you think it's worth making this a subtest that runs the same set of requests against both `true`/`false` and asserts on the status codes appropriate for either?

That way at least we've got one test that should cover both configurations, rather than relying on existing tests to be covering `false` status codes

Reply:
I planned to just rely on the existing tests, but I can make this a subtest and assert on the proper status codes for not forcing API forbidden errors.